src/pages/Settings/TwoStepVerificationPage.vue

```diff
@@ -38,33 +93,143 @@
 import TheMyKivaSecondaryMenu from '@/components/WwwFrame/Menus/TheMyKivaSecondaryMenu';
 import WwwPage from '@/components/WwwFrame/WwwPage';
+import removeMfa from '@/graphql/mutation/removeMfa.graphql';
+
+const mfaQuery = gql`query mfaQuery($mfa_token: String!) {
+	my {
+		lastLoginTimestamp @client
+		authenticatorEnrollments(mfa_token: $mfa_token) {
+			id
+			active
+			authenticator_type
+		}
+	}
+}`;
 
 export default {
+	data() {
+		return {
+			isMfaActive: false,
+			lastLoginTime: 0,
+		};
+	},
 	components: {
 		KvIcon,
 		KvSettingsCard,
+		KvButton,
 		TheMyKivaSecondaryMenu,
 		WwwPage,
 	},
 	metaInfo: {
 		title: '2-step verification',
 	},
+	mounted() {
+		if (this.kvAuth0.enabled) {
+			this.kvAuth0.checkSession();
+			this.kvAuth0.getMfaManagementToken()
+				.then(token => {
+					return this.apollo.query({
+						query: mfaQuery,
+						variables: {
+							mfa_token: token
+						}
+					});
+				}).then(result => {
+					const authEnrollments = result.data.my.authenticatorEnrollments;
+					this.lastLoginTime = result.data.my.lastLoginTimestamp;
+					for (let i = 0; i < authEnrollments.length; i += 1) {
+						// eslint-disable-next-line max-len
+						if (authEnrollments[i].active === true && authEnrollments[i].authenticator_type !== 'recovery-code') {
+							this.isMfaActive = true;
+							return;
+						}
+					}
+				});
+		}
+		if (this.$route.query.mfa === 'off') {
+			// User returns to page after login, or if has logged in within 5 minutes
+			// and is presented with a window.confirm
+			const mfaOffConfirm = window.confirm('Are you sure you want to turn off 2-step verification?');
+			if (mfaOffConfirm) {
+				// Upon confirm triggger mutation to turn off mfa
+				this.turnOffMfa();
+			} else {
+				// Upon cancel return to the base URL of current page
+				window.location = '/settings/security/mfa';
+			}
+		}
+	},
+	inject: ['apollo', 'kvAuth0'],
+	computed: {
+		cardTitle() {
+			if (this.isMfaActive) {
+				return 'Add a backup method';
+			}
+			return 'How would you like your verification code?';
+		},
+		cardSubhead() {
+			if (this.isMfaActive) {
+				return 'Set up additional backup steps so you can log in even if your other options aren\'t available';
+			}
+			return 'You\'ll be asked for a verification code when accessing you Kiva account.';
+		}
+	},
+	methods: {
+		checkLastLoginTime() {
+			const timeSinceLastLogin = (Math.floor(Date.now()) - this.lastLoginTime) / 60 / 1000;
+			const doneUrl = `${this.$route.path}?mfa=off`;
+			const encodedDoneUrl = encodeURIComponent(doneUrl);
+
+			/* If last login was more than 5 minutes ago, send user to login
+				otherwise update the URL to include ?mfa=off, which triggers
+				a condition in the mounted hook */
+			if (timeSinceLastLogin >= 5) {
+				window.location = `/ui-login?force=true&doneUrl=${encodedDoneUrl}`;
+			} else {
+				window.location = doneUrl;
+			}
+		},
+		turnOffMfa(token) {
+			this.apollo.mutate({
```

Comment:
Instead of taking a token as an argument, you'll need to fetch it again, like in my comment above
```
this.kvAuth0.checkSession()
	.then(() => this.kvAuth0.getMfaManagementToken())
	.then(token => {
		this.apollo.mutate({
```